Sources/Common/DataModel/Box/index.d.ts

```diff
@@ -26,6 +31,16 @@ export interface vtkBox extends vtkObject {
      */
     evaluateFunction(x: Vector3): number;
 
+    /**
+     * Intersect box with line and return the parametric values and points of the two intercepts
+     * @param bounds 
+     * @param p1 
+     * @param p2 
+     * returns @object IBoxIntersections {t1, t2, x1, x2} object containing the t1, t2 parametric values and
+     * x1, x2 coordinates of the line intercept points in the bounding box
+     */
+    intersectWithLine(p1: Vector3, p2: Vector3) : IBoxIntersections;
```

Comment:
you said earlier than it can return undefined, so you need to add that possibility